org.eclipse.ice.dev/org.eclipse.ice.dev.annotations/src/main/resources/templates/DataElement.vm

```diff
@@ -110,6 +124,19 @@ public class ${class} implements ${interface}, Serializable {
 		}
 	}
 
+	/**
+	 * Copy the contents of another element into this element.
+	 * @param element the element to copy
+	 */
+	@Override
+	public void copy(@NonNull $interface element) {
```

Comment:
I'm guessing the `@NonNull` flag is a check that handles the null exceptions you previously had in copy?

Reply:
It's not too obvious from the template but I realized that some of the previous checks in the copy constructor were pointless, i.e. I had an `instanceof` check that literally amounted to `if (generatedDataElementImplementationInstance instanceof GeneratedDataElementImplementation) ...`. So removing that, the only thing left was a null check that threw a generic exception which I also realized was no different from the null pointer exception that is thrown by the [`@NonNull` annotation](https://projectlombok.org/features/NonNull).